numpyro/infer/sa.py

```diff
@@ -258,6 +259,7 @@ def _init_state(self, rng_key, model_args, model_kwargs, init_params):
             init_params = init_params[0]
             # NB: init args is different from HMC
             self._init_fn, sample_fn = _sa(potential_fn_gen=potential_fn)
+            self._potential_fn_gen = potential_fn
```

Comment:
lgtm but why `_gen`?

Reply:
This creates a potential_fn given inputs args/kwargs, hence the name. We distinguished `potential_fn` and `potential_fn_gen` in `hmc` implementation.